srcs/app/http/FormFile.cpp

```diff
@@ -0,0 +1,205 @@
+#include <FormFile.hpp>
+#include <stdexcept>
+#include <StringUtils.hpp>
+
+const char FormFile::kCRLF[] = "\r\n";
+const char FormFile::kWhitespace[] = " \t";
+
+FormFile::FormFile(const HttpRequest &request) {
+	ParseRequestContentType_(request);
+	ParseRequestBody_(request);
+}
+
+std::string	FormFile::GetFilename() const {
+	return filename_;
+}
+
+std::string	FormFile::GetFileContent() const {
+	return file_content_;
+}
+
+// Syntax: Content-Type: multipart/form-data; boundary=something
+void	FormFile::ParseRequestContentType_(const HttpRequest &request) {
+	if (!request.HasHeader("Content-Type")) {
+		throw std::invalid_argument("[FormFile] Invalid request Content-Type");
+	}
+	const std::string content_type = request.GetHeaderValue("Content-Type");
+	std::size_t index = ParseMediaType_(content_type, 0, "multipart/form-data");
+	ParseBoundary_(content_type, index);
+}
+
+/*
+** Example body
+--something\r\n
+Content-Disposition: form-data; name="filename"; filename="foo.txt"\r\n
+Content-Type: text/plain\r\n
+\r\n
+foobar\n\r\n
+--something--\r\n
+*/
+void	FormFile::ParseRequestBody_(const HttpRequest &request) {
```

Comment:
I think this method is only reading the first field of the form?🤔

Reply:
You are correct. This method is reading the first field of the form, and it requires to be a file. If it has more than one field or is not a file, it throws an error.